src/main/java/cslicer/analyzer/Slicer.java

```diff
@@ -676,6 +672,106 @@ public int compare(Set<RevCommit> o1, Set<RevCommit> o2) {
 	}
 
 	/**
+	 * 
+	 *
+	 *
+	 * @return the string command to run for the pull-request
+	 */
+	public String getPullRequestCommand(){
+		String commandString =
+				"git push upstream"+" && " +
+				"hub pull-request -h $head:VERIFYTEST -b $base:$basebranch -m SingleFeature";
```

Comment:
We should minimize the usage of command-line dependencies (users may not have these tools installed). Please checkout the egit library (http://javadox.com/org.eclipse.mylyn.github/org.eclipse.egit.github.core/2.1.4/org/eclipse/egit/github/core/service/PullRequestService.html) and this post (https://stackoverflow.com/questions/53802733/github-api-createpullrequest-java-api)

I think both pull request creation and "push upstream" can be done through these APIs.